### PR TITLE
Feature/python model beta

### PR DESCRIPTION
**Pull request by ueshin**
### Description

Ports the changes for python model v1 from [`dbt-spark`](https://github.com/dbt-labs/dbt-spark/pull/377) but use APIs below instead.

- [Create an execution context](https://docs.databricks.com/dev-tools/api/1.2/index.html#create-an-execution-context
)
- [Run a command](https://docs.databricks.com/dev-tools/api/1.2/index.html#run-a-command)
- [Get information about a command](https://docs.databricks.com/dev-tools/api/1.2/index.html#get-information-about-a-command)
  - loop until the command ends
- [Delete an execution context](https://docs.databricks.com/dev-tools/api/1.2/index.html#delete-an-execution-context)

### CODE REVIEW

**Comment by ueshin** [comment]
This is now based on #148. Please take a look at it first.

File: dbt/adapters/databricks/impl.py [hunk 1]
```diff
@@ -214,6 +225,74 @@ def _get_columns_for_catalog(self, relation: DatabricksRelation) -> Iterable[Dic
...
+                    time.sleep(1)
+                    json_resp = api_client.Command.status(
+                        cluster_id=cluster_id, context_id=context_id, command_id=command_id
+                    )
+                    status = json_resp["status"]
+                    # logger.debug(f"Polling.... in state: {state}")
+                if status != "Finished":
+                    raise dbt.exceptions.RuntimeException(
+                        "python model run ended in state"
+                        f"{status} with state_message\n{json_resp['results']['data']}"
```
Comment (ChenyuLInx) [inline dbt/adapters/databricks/impl.py:285]: I noticed that if the python code actually failed, status will still be finish, but `json_resp['results']['resultType']` will be `error`, and `json_resp['results']['cause']` will contain traceback. I reused your code for jobs API and pushed the changes [here](https://github.com/dbt-labs/dbt-spark/pull/442/files). Feel free to review and let me know if any adjustment needed!

File: dbt/adapters/databricks/api_client.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+
+class Api12Client:
+    def __init__(self, host: str, token: str, command_name: str = ""):
+        self._api_client = ApiClient(
+            user="token",
+            password=token,
+            host=f"https://{host}",
+            api_version="1.2",
+            command_name=command_name,
```
Comment (ChenyuLInx) [inline dbt/adapters/databricks/api_client.py:14]: Is this `command_name` being recorded somewhere after the run?

File: dbt/adapters/databricks/impl.py (hunk 1 above)
Comment (ueshin) [reply dbt/adapters/databricks/impl.py:285]: Thanks! I'll address it.

File: dbt/adapters/databricks/api_client.py (hunk 2 above)
Comment (ueshin) [reply dbt/adapters/databricks/api_client.py:14]: It will be part of user agent in the API request.

Comment (ueshin) [reply dbt/adapters/databricks/api_client.py:14]: @ChenyuLInx @jtcohen6 btw, is it possible to set a simple string in the user agent of `dbt-spark` to identify the API calls are from `dbt` similar to ODBC connections?

Comment (ChenyuLInx) [reply dbt/adapters/databricks/api_client.py:13]: @jtcohen6  I am not very familiar with what we did with ODBC connections, so I will defer it to you.
@ueshin feel free to open an issue in dbt-spark about this!

**Comment by zhengruifeng** [comment]
https://github.com/databricks/jaffle_shop/pull/1/files#r969108078
please take a look at this, it seems that python model cann't work with `view` materialization

**Comment by ueshin** [comment]
> it seems that python model cann't work with `view` materialization

Yes, so far it's a limitation of python model.
https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models#materializations

**Comment by zhengruifeng** [comment]
@ueshin Since it's a known issue, then I think this PR works well. I don't find other issue right now

**Review by zhengruifeng — APPROVED** [review]

File: dbt/adapters/databricks/api_client.py (hunk 2 above)
Comment (ueshin) [reply dbt/adapters/databricks/api_client.py:13]: @ChenyuLInx I opened an issue in dbt-spark https://github.com/dbt-labs/dbt-spark/issues/460. Thanks.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**